### PR TITLE
Add sonar scan

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>

https://github.com/stolostron/backlog/issues/23658

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,10 @@
+sonar.projectKey=open-cluster-management_registration
+sonar.projectName=registration
+sonar.sources=.
+sonar.exclusions=**/*_test.go,**/vendor/**,/test/**
```
Comment (qiujian16) [inline sonar-project.properties]: you may also want to exclude cmd

Comment (xuezhaojun) [reply sonar-project.properties]: cmd exclusion added.

**Pull request merged**